pkg/booleanpolicy/policyfields/fields.go

```diff
@@ -106,7 +106,9 @@ func ContainsMemResourceLimit(p *storage.Policy) bool {
 	return booleanpolicy.ContainsValueWithFieldName(p, fieldnames.ContainerMemLimit)
 }
 
-// ContainsUnscannedImageField returns whether the policy contains the unscanned image field.
-func ContainsUnscannedImageField(p *storage.Policy) bool {
-	return booleanpolicy.ContainsValueWithFieldName(p, fieldnames.UnscannedImage)
+// ContainsScanRelatedFields returns whether the policy contains fields related to image scanning,
+// i.e. fieldnames.UnscannedImage or fieldnames.ImageSignatureVerifiedBy.
+func ContainsScanRelatedFields(p *storage.Policy) bool {
+	return booleanpolicy.ContainsValueWithFieldName(p, fieldnames.UnscannedImage) ||
+		booleanpolicy.ContainsValueWithFieldName(p, fieldnames.ImageSignatureVerifiedBy)
```

Comment:
This list isn't really complete, right now, only `fieldnames.UnscannedImage, fieldnames.ImageSignatureVerifiedBy` are used, there's also `fieldnames.ImageScanAge`.
Since I'm anyways touching this, would it make sense to also add the scan age one here?

Reply:
I would not add the other ones, but it does make sense for ImageSignatureVerifiedBy. Just note that if the scan/signature cannot be pulled then it will be fail open based on the current code. This is actually how it is for the other ones as well if the scan is unsuccessful, just by the nature of the scan being nil so things like image scan age, etc cannot be checked

Reply:
Yeah, we should rename the function to be more clear. These are not just scan related fields, they are a specific kind of field where the policy might fail _without_ a scan but pass _after_ a scan is received. All the other scan related fields (CVE, scan age) are such that getting a scan makes the policy strictly more likely to fail.